### PR TITLE
Make infra/version_check.py only run on branches and tags, not PRs

### LINKED ISSUE #271: Make infra/version_check.py only run on branches and tags, not pull requests
It is painful for contributors otherwise.

### PR DESCRIPTION
**Pull request by sbeesm**
Fixes #271.

### CODE REVIEW

**Comment by sbeesm** [comment]
> Submitting the task to Taskcluster failed. Details
> TemplateError at template.tasks[0].payload: $if has undefined properties: image maxRunTime

@marco-c sorry, I'm not familiar with `yml`.

Could you please tell me what's wrong here and how to fix this?

**Comment by marco-c** [comment]
I'm not so familiar with the format either, you'll have to use the docs https://taskcluster-web.netlify.com/docs/reference/integrations/github/taskcluster-yml-v1 and a bit of trial and error.

**Comment by sbeesm** [comment]
I've defined a separate task that runs the script when task is github-push.

Although the checks have passed, but how can I verify that it does the right thing?

File: .taskcluster.yml [hunk 1]
```diff
@@ -45,14 +45,38 @@ tasks:
...
+            maxRunTime: 3600
+            image: python
+            command:
+              - "/bin/bash"
+              - "-lcx"
+              - "git clone ${repository} &&
+                 cd bugbug &&
+                 git checkout ${head_rev} &&
+                 pip install -r test-requirements.txt &&
+                 flake8 &&
```
Comment (marco-c) [inline .taskcluster.yml]: This task won't need to run flake8, given that we already run it in the linting task.

File: .taskcluster.yml [hunk 2]
```diff
@@ -45,14 +45,38 @@ tasks:
...
+               flake8"
         metadata:
           name: bugbug lint
           description: bugbug lint
           owner: ${user}@users.noreply.github.com
           source: ${repository}/raw/${head_rev}/.taskcluster.yml
 
+      - $if: 'tasks_for == "github-push"'
+        then:
+          taskId: {$eval: as_slugid("version_check_task")}
```
Comment (marco-c) [inline .taskcluster.yml]: This task should be added as a dependency of the release task.

**Comment by marco-c** [comment]
> Although the checks have passed, but how can I verify that it does the right thing?

After merging, we'll verify by looking at the Taskcluster logs of the tasks on master.

File: .taskcluster.yml (hunk 2 above)
Comment (sbeesm) [reply .taskcluster.yml]: So do you mean to define this task in the release task itself or just adding the dependency like we've done for other tasks?

Comment (marco-c) [reply .taskcluster.yml]: Just the dependency

File: .taskcluster.yml [hunk 3]
```diff
@@ -45,14 +45,37 @@ tasks:
...
+          payload:
+            maxRunTime: 3600
+            image: python
+            command:
+              - "/bin/bash"
+              - "-lcx"
+              - "git clone ${repository} &&
+                 cd bugbug &&
+                 git checkout ${head_rev} &&
+                 pip install -r test-requirements.txt &&
```
Comment (marco-c) [inline .taskcluster.yml]: We don't need to install the test-requirements in this case

File: .taskcluster.yml [hunk 4]
```diff
@@ -45,14 +45,37 @@ tasks:
...
+                 python infra/version_check.py"
+          metadata:
+            name: bugbug version_check
+            description: bugbug version_check
```
[the remark this answers is not in the document]
Comment (sbeesm) [reply .taskcluster.yml]: Should we do this for `taskId` as well?

[the remark this answers is not in the document]
Comment (marco-c) [reply .taskcluster.yml]: No the taskId is OK

**Pull request merged**